testsuite/integration/web/src/test/java/org/jboss/as/test/integration/web/handlers/RequestDumpingHandlerTestCase.java

```diff
@@ -0,0 +1,299 @@
+package org.jboss.as.test.integration.web.handlers;
+
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.PERIODIC_ROTATING_FILE_HANDLER;
+import static org.jboss.as.controller.descriptions.ModelDescriptionConstants.SUBSYSTEM;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Scanner;
+import java.util.regex.Pattern;
+
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.as.controller.operations.common.Util;
+import org.jboss.as.test.integration.management.ManagementOperations;
+import org.jboss.dmr.ModelNode;
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Tests the use of undertow-handlers.conf
+ *
+ * @author <a href="mailto:jstourac@redhat.com">Jan Stourac</a>
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class RequestDumpingHandlerTestCase {
+
+    private static Logger log = Logger.getLogger(RequestDumpingHandlerTestCase.class);
+
+    /** Name of the log file that will be used for testing. */
+    private static String logFilePrefix;
+    private static final String LOG_FILE_SUFFIX = ".log";
+
+    @ArquillianResource
+    private static ManagementClient mgmtClient;
+
+    /** Path to custom server log file. */
+    private static Path logFilePath;
+
+    private static ModelNode originalValue;
+    private static String relativeTo;
+
+    // Address to server log setting
+    private static final PathAddress aLogAddr = PathAddress.pathAddress().append(SUBSYSTEM, "logging")
+            .append(PERIODIC_ROTATING_FILE_HANDLER, "FILE");
+
+    @Before
+    public void setup() throws Exception {
```

Comment:
it would be cleaner if you would use ServerSetupTask for this kind of things. most of tests use it.
you just add annotation @ServerSetup(name-of-your-setup-class.class)
and you get access to mgmt client and friends.